tests/run/cpp_stl_string_view.pyx

```diff
@@ -35,6 +35,30 @@ def test_return_to_py():
     return return_to_py()
 
 
+def test_coercion_py(py_str):
+    """
+    >>> b_asdf_5 = b_asdf * 5
+    >>> test_coercion_py(b_asdf_5)
+    """
+    if not isinstance(py_str, bytes):
+        return
+
+    cdef string_view sv1 = py_str
+    assert len(py_str) == sv1.size()
+    assert <const char*>py_str == sv1.data()
+
+    cdef const char[:] mv1 = sv1
```

Comment:
Looks like this case fails

Was hoping it would do something equivalent to

```cython
cdef const char[:] mv1 = <char[:sv1.length():1]><const char*>sv1.data()
```

However it looks like this doesn't work for `std::string` either. Plus this is a non-owning instance. So it makes sense why we might want to add more hoops to jump through for this case

Reply:
> `cdef const char[:] mv1 = <char[:sv1.length():1]><const char*>sv1.data()`

Shouldn't that be

```
cdef const char[:] mv1 = <const char[:sv1.length():1]><const char*>sv1.data()
```

?

I think it's reasonable to jump through hoops for this and strings because the memoryview can't own them

Reply:
Both seemed to work, but agree `const char` would be better

Excluding this case makes sense. Didn't quite follow the last bit